### PR TITLE
Split edge_partition_src_dst_property.cuh to .hpp and .cuh files.

### PR DESCRIPTION
**Pull request by seunghwak**
Partially addresses #2479
Code clean-up before adding `edge_property_t` and `update_edge_property` (to support edge property in addition to edge weights).

- [x] Split `edge_partition_src_dst_property.cuh` to `edge_src_dst_property.hpp` (can be complied with g++) and `edge_partition_endpoint_device_view.cuh` (requires nvcc)
- [x] Rename `edge_partition_src|dst_property_t` to `edge_src|dst_property_t`
- [x] Update `update_edge_partition_src_dst_property.cuh` to `update_edge_src_dst_property.cuh` and `update_edge_partition_src|dst_property` to `update_edge_src|dst_property`.
- [x] Rename `dummy_property_t` to `edge_src_dummy_property_t` and `edge_dst_dummy_property_t`
- [x] Replaced `edge_src|dst_property_t`'s `fill()` with `fill_edge_src|dst_property` (to allow including `edge_src_dst_property.hpp to be compiled with g++).
- [x] Created `edge_src|dst_property_view_t` classes (host side, in addition to the device_view objects which can be trivially copied to the device side); so the view() member function can be compiled with g++.

### CODE REVIEW

**Comment by seunghwak** [comment]
rerun tests

**Comment by seunghwak** [comment]
rerun tests

File: cpp/src/prims/fill_edge_src_dst_property.cuh [hunk 1]
```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2020-2022, NVIDIA CORPORATION.
```
Comment (ChuckHastings) [inline cpp/src/prims/fill_edge_src_dst_property.cuh:2]: Shouldn't this just be 2022?

**Review by ChuckHastings — COMMENTED** [review]
LGTM except a couple of copyright questions.

File: cpp/include/cugraph/edge_partition_endpoint_property_device_view.cuh [hunk 2]
```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2021-2022, NVIDIA CORPORATION.
```
[the remark this answers is not in the document]
Comment (seunghwak) [reply cpp/include/cugraph/edge_partition_endpoint_property_device_view.cuh:2]: I am not sure.

This file is a new file, in that sense, this should be 2022.

But the contents of this file is from another older file. That file is split to two files (and after split, significant updates, but I guess this is irrelevant in this discussion). So, in this case, the two new files should have the copyright year of just 2022 or need to inherit the copyright year of the original file? I thought the latter but will appreciate clarification.

File: cpp/src/prims/fill_edge_src_dst_property.cuh (hunk 1 above)
Comment (seunghwak) [reply cpp/src/prims/fill_edge_src_dst_property.cuh:2]: Yes, this should be 2022. I will make a fix.

File: cpp/include/cugraph/edge_partition_endpoint_property_device_view.cuh (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChuckHastings) [reply cpp/include/cugraph/edge_partition_endpoint_property_device_view.cuh:2]: That wasn't clear when reviewing.  I would agree that the contents of a file that originated in 2021 and is split into two files would need to keep 2021 as part of the copyright definition.

**Review by ChuckHastings — APPROVED** [review]

**Review by naimnv — APPROVED** [review]

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**